Tests/ReactiveSwiftTests/PropertySpec.swift

```diff
@@ -819,6 +819,64 @@ class PropertySpec: QuickSpec {
 				}
 			}
 
+			describe("lens") {
+				it("should return the result of the getter on each value change") {
+					let initialValue = (character: "🎃", other: 42)
+					let nextValue = (character: "😾", other: 74)
+
+					let tupleProperty = MutableProperty<(character: String, other: Int)>(initialValue)
+					let theLens = tupleProperty.lens { $0.character }
+
+					expect(theLens).to(sendValue("🎃", sendError: nil, complete: false))
+
+					tupleProperty.value = nextValue
+
+					expect(theLens).to(sendValue("😾", sendError: nil, complete: false))
+				}
+
+				it("should evaluate its getter lazily") {
+					let initialValue = (character: "🎃", other: 42)
+					let tupleProperty = MutableProperty<(character: String, other: Int)>(initialValue)
+
+					var getterEvaluated = false
+					let theLens = tupleProperty.lens { (tuple: (character: String, other: Int)) -> String in
+						getterEvaluated = true
+						return tuple.character
+					}
+
+					expect(getterEvaluated).to(beFalse())
+					expect(theLens).to(sendValue("🎃", sendError: nil, complete: false))
+					expect(getterEvaluated).to(beTrue())
+				}
+
+				it("should evaluate its getter lazily on the specified scheduler") {
+					let initialValue = (character: "🎃", other: 42)
+					let tupleProperty = MutableProperty<(character: String, other: Int)>(initialValue)
+
+					let testScheduler = TestScheduler()
+
+					var getterEvaluated = false
+					let theLens = tupleProperty.lens(on: testScheduler) { (tuple: (character: String, other: Int)) -> String in
+						getterEvaluated = true
+						return tuple.character
+					}
+
+					var characters: [String] = []
+					theLens.startWithValues { character in
+						characters.append(character)
+					}
+
+					expect(getterEvaluated).to(beFalse())
+					expect(characters).to(beEmpty())
+
+					testScheduler.run()
```

Comment:
> "the getter should execute on the specific scheduler/thread that is specified"

Yeah, that's exactly what I was trying to verify. I agree that in this case having a simpler test is better, but I admin I got caught up in trying to solve the problem of verifying that execution occurs on the expected queue/scheduler. 😛

I believe that code does what it says on the tin (and is thread-safe), so if you think it's valuable to include it feel free to use it as inspiration. Some of those details deserve to be factored out into some kind of test helper anyway, so that the test itself can be higher level.